ambari-server/src/main/java/org/apache/ambari/server/stack/StackServiceDirectory.java

```diff
@@ -101,8 +103,17 @@ public String getAdvisorName(String serviceName) {
     String stackName = stackDir.getName();
     String versionString = stackVersionDir.getName().replaceAll("\\.", "");
 
+    List<String> legacyStackNames = new ArrayList<>(Arrays.asList("HDP", "HDF")); // Old stacks.
```

Comment:
We shouldn't hardcode stack names. What about IOP? We need to handle this without hardcoding.

Reply:
Please file a separate follow up JIRA and we can see how to deal with this. Lets not block this change for now.

Reply:
JIRA created: https://issues.apache.org/jira/browse/AMBARI-23905